app/renderer/js/main.ts

```diff
@@ -410,10 +412,16 @@ class ServerManagerView {
 			this.openSettings('AddServer');
 		});
 		this.$settingsButton.addEventListener('click', () => {
+			this.previousTabIndex = this.tabs[0].props.index;
 			this.openSettings('General');
 		});
 		this.$backButton.addEventListener('click', () => {
-			this.tabs[this.activeTabIndex].webview.back();
+			if (this.tabs[this.activeTabIndex].webview.props.name === 'Settings') {
+				this.tabs[this.activeTabIndex].deactivate();
+				this.activateTab(this.previousTabIndex);
+			} else {
+				this.tabs[this.activeTabIndex].webview.back();
+			}
```

Comment:
This logic has some problems. What if there is no previous tab? What if the previous tab is also the Settings tab? Empirically, if I open the app with a freshly deleted configuration and the first thing I click on is the Back button, I end up at a white screen from which neither the Settings button nor the Add organization button do anything.

Reply:
@andersk I need to fix the issue from the beginning so that it will take care of all the test cases. Thanks for pointing me out about this case.